spec/signalfx-transform-helper-spec.js

```diff
@@ -1,81 +1,44 @@
-    const flattened = signalfxHelper.toKeyValueMap(obj);
-    expect(flattened.IP_Address).to.equal("123.45.67.89");
-    expect(flattened.field_nested_object_name).to.equal('test');
+  it('should sanitize keys', () => {
+    const obj = {'!ke@£y $%^1&': 'value'};
+    const expected = {detail_key1: 'value'};
+    const sanitizedAndPrefixed = signalfxHelper.transformDetails(obj);
+    expect(sanitizedAndPrefixed).to.deep.equal(expected);
   });
 
-  it('should flatten nested properties', () => {
-    const obj = { key: { nestedKey: 'value', otherNestedKey: 'otherValue'}};
-    const flattened = signalfxHelper.toKeyValueMap(obj);
-    expect(flattened['key_nestedKey']).to.be.equal('value');
-    expect(flattened['key_otherNestedKey']).to.be.equal('otherValue');
-  });
-
-  it('should handle different levels of nesting', () => {
+  it('should leave flat objects as they are', () => {
     const obj = {
-      testA: 'a',
-      testB: { testBB: 1},
-      testC: { testCC: {testCCC: 'c', testCCCC: 'cccc'}},
-      testD: { testD: 2, testDD: 'dd'}
+      key1: 'value',
+      key2: 5
     };
-    const flattened = signalfxHelper.toKeyValueMap(obj);
-    expect(flattened['testA']).to.be.equal('a');
-    expect(flattened['testB_testBB']).to.be.equal(1);
-    expect(flattened['testC_testCC_testCCC']).to.be.equal('c');
-    expect(flattened['testC_testCC_testCCCC']).to.be.equal('cccc');
-    expect(flattened['testD_testD']).to.be.equal(2);
-    expect(flattened['testD_testDD']).to.be.equal('dd');
-  });
-
-  it('should handle arrays', () => {
-    const obj = {
-      resources: ['a', 'b', 'c']
+    const expected = {
+      detail_key1: 'value',
+      detail_key2: 5
     };
-    const flattened = signalfxHelper.toKeyValueMap(obj);
-    expect(flattened['resources_0']).to.be.equal('a');
-    expect(flattened['resources_1']).to.be.equal('b');
-    expect(flattened['resources_2']).to.be.equal('c');
+    const sanitizedAndPrefixed = signalfxHelper.transformDetails(obj);
+    expect(sanitizedAndPrefixed).to.deep.equal(expected);
   });
 
-  it('should handle arrays of objects', () => {
+  it('should stringify arrays and objects', () => {
     const obj = {
-      resources: ['a', {b: 'B'}, {c: {c: {c: 'C'}}}, 'd', {e: [1,2,3]}]
+      key1: 'value',
+      key2: 5,
+      key3: {inner1: 42, inner2: 43},
+      key4: [2, "one"]
     };
-    const flattened = signalfxHelper.toKeyValueMap(obj);
-    expect(flattened['resources_0']).to.be.equal('a');
-    expect(flattened['resources_1_b']).to.be.equal('B');
-    expect(flattened['resources_2_c_c_c']).to.be.equal('C');
-    expect(flattened['resources_3']).to.be.equal('d');
-    expect(flattened['resources_4_e_0']).to.be.equal(1);
-    expect(flattened['resources_4_e_1']).to.be.equal(2);
-    expect(flattened['resources_4_e_2']).to.be.equal(3);
-  });
-
-  it('should handle sample Amazon event', () => {
-    const event = require('./amazonAlert.json');
```

Comment:
if that tests gets thrown out, we can probably remove the json file as well? please check if it's used anywhere else and remove if it's not.